src/core/service.c

```diff
@@ -2877,6 +2918,10 @@ static int service_stop(Unit *u) {
                 service_set_state(s, service_determine_dead_state(s));
                 return 0;
 
+        case SERVICE_MOUNTING:
+                service_kill_control_process(s);
+                service_mount_request_reply(s, false, SD_BUS_ERROR_DISCONNECTED);
```

Comment:
prefix 2nd param with `/* success= */`

Reply:
SD_BUS_ERROR_DISCONNECTED is about bus connections, clients often consider the connection dead when they see this. Just add a proper error code for this: "org.freedesktop.systemd1.MountCancelled" or so. add to src/libsystemd/sd-bus/bus-common-errors.[ch]

Reply:
Actually I'll switch to BUS_ERROR_UNIT_INACTIVE here, as that's the problem: the unit went inactive, so it seems more useful than a generic "cancelled", as it gives a more specific indication of what the reason was